### DISCUSSION TITLE
tools/fastdds/fastdds needs shebang

### DISCUSSION THREAD

**Discussion by SubaruArai** [eProsima/Fast-DDS#3734] [Ideas]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

The scripts should run regardless of the shell used.

### Current behavior

It doesn't run with non-POSIX compliant shells (e.g. fish)

### Steps to reproduce

1. start a `fish` shell (or any non-POSIX compliant shells, that doesn't support `var=xxx`)
2. run `fastdds` or `ros-discovery`
3. see how it fails

### Fast DDS version/commit

All 2.x.x versions

### Platform/Architecture

Ubuntu Focal 20.04 amd64, Ubuntu Focal 20.04 arm64

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

It is solvable by adding following shebang:
```sh
#! /usr/bin/env sh
```

To those 2 files:
- [tools/fastdds/fastdds](tools/fastdds/fastdds)
- [tools/fastdds/ros-discovery](tools/fastdds/ros-discovery)

**Comment by SubaruArai** [comment]
I can't sign the DCO anytime soon, so I posted as an issue.

**Comment by JLBuenoLopez** [comment]
Thanks for your report @SubaruArai,

I have issued #3732 to improve the scripts portability.

Regarding the DCO, the only requirement is that your commit is signed which is easy to do using the option `-s` when committing: `git commit -s`. So next time you can contribute yourself if you want.

I have labeled this feature as enhancement and consequently I am moving the ticket to the Ideas discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md)

**Reply by SubaruArai** [reply]
Thanks!

> Regarding the DCO...

Yeah, it's easy to sign. But it's more of a company policy problem on my side 😄 .
